lib/astarte/core/generators/device.ex

```diff
@@ -0,0 +1,163 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Core.Generators.Device do
+  @moduledoc """
+  This module provides generators for Astarte Device.
+
+  See https://hexdocs.pm/astarte_core/Astarte.Core.Device.html
+
+  """
+  use ExUnitProperties
+  alias Astarte.Common.Generators.Ip, as: IpGenerator
+  alias Astarte.Common.Generators.Timestamp, as: TimestampGenerator
+  alias Astarte.Core.Device
+  alias Astarte.Core.Interface
+
+  @doc """
+  Generates a valid Astarte Device
+  """
+  @spec device([{:interfaces, Interface.t()}, ...]) :: StreamData.t(map())
```

Comment:
Nit: device introspection could also be empty, as it is possible for a device to connect and not send any data